pkg/ingress/query.go

```diff
@@ -150,3 +158,41 @@ func getRule(host string, path string, pathType *networkingv1.PathType, backend
 	}
 	return rule
 }
+
+func getIngressClass() (string, bool) {
+	ingressClass := ""
+	nginxIngressAvailable := false
+	// This statement only work if it is running inside cluster, if you want to test outside replace this statement with
+	// config, err := clientcmd.BuildConfigFromFlags("", "PATH_TO_YOUR_CLUSTER_CONFIG_FILE")
+	config, err := rest.InClusterConfig()
+	if err != nil {
+		return "", true
+	}
+	clientSet, err := kubernetes.NewForConfig(config)
+	if err != nil {
+		return "", true
+	}
+	ingressList, err := clientSet.NetworkingV1().IngressClasses().List(context.Background(), metav1.ListOptions{})
+	if err != nil {
+		return "", true
+	}
+	for _, ingress := range ingressList.Items {
+		if ingress.Name == "nginx" {
+			nginxIngressAvailable = true
+		}
+		for k, v := range ingress.Annotations {
+			if k == "ingressclass.kubernetes.io/is-default-class" {
+				if v == "true" {
+					ingressClass = ingress.Name
+					break
+				}
+			}
+		}
+	}
+	if len(ingressClass) > 0 {
+		return ingressClass, false
+	} else if nginxIngressAvailable {
+		return "nginx", false
+	}
+	return "", true
```

Comment:
Shouldn't this be `false`?

Reply:
This is if there is neither default ingress controller is there nor nginx ingress controller available.